### ISSUE TITLE
Insufficient library dependency handling

### ISSUE THREAD

**Issue by wadoon** [KeYProject/key#1073]
This issue was created at [git.key-project.org](https://git.key-project.org/key/key/-/issues/1073) where the discussions are preserved.

                ----

                * Mantis: [MT-1391](http://i12www.ira.uka.de/~klebanov/mantis/view.php?id=1391)
* Submitted on: 2013-12-20 by @klebanov
* Updated: 2013-12-21

* Assigned to: @rbubel

### Description

> It is currently not possible (without contortions) to have older and newer key versions installed in parallel since they require different versions of antlr.jar.
> 
> Possible solutions I can think of:
> - Track library versions, i.e., demand antlr-3.5.1.jar
> - Extend the scripts to handle mixed key-local and shared library installation

## Notes

_@mulbrich at 2013-12-20_
> I understand that the ANTLR runtime lib that we use contains both 2.7.2 and 3.5.1.
> 
> Apparently "antlr-3.5.1-complete.jar" can be used for all KeY versions.

_@klebanov at 2013-12-20_
> Thank you for pointing this out. Yet, when trying to compile KeY 1.6 with that antlr, I receive a bunch of compilation errors of the form
> 
> genSrc/de/uka/ilkd/key/parser/KeYLexer.java:591: error: cannot find symbol
>                                 recover(ex,_tokenSet_0);
>                                 ^
>   symbol:   method recover(RecognitionException,BitSet)
>   location: class KeYLexer
> genSrc/de/uka/ilkd/key/parser/KeYLexer.java:614: error: cannot find symbol
>                                 recover(ex,_tokenSet_0);
>                                 ^
>   symbol:   method recover(RecognitionException,BitSet)
>   location: class KeYLexer
> genSrc/de/uka/ilkd/key/parser/KeYLexer.java:637: error: cannot find symbol
>                                 recover(ex,_tokenSet_0);
> 
> [...]

_@rbubel at 2013-12-20_
> Hi,
> 
>  antlr 3.5.1 complete contains antlr 2.7.7. To make the old key parser compile with antler 2.7.7 you need to add the following two lines to keyparser.g respective lexer.g
> 
> Then it should compile and work without problem.
> 
> Happy holidays!
> 
> keyparser.g:
> 
>   public void recover( RecognitionException ex, BitSet tokenSet ) throws TokenStreamException {
>      consume();
>      consumeUntil( tokenSet );
>     }
> 
>  lexer:
>    public void recover( RecognitionException ex, BitSet tokenSet ) throws CharStreamException {
>      consume();
>      consumeUntil( tokenSet );
>    }

_@klebanov at 2013-12-20_
> Thanks! I will try this later.

_@klebanov at 2013-12-20_
> That worked!
> 
> From my side, the occasion for opening this bug has disappeared, so feel free to resolve/won't fix it.
> 
> Happy holidays!

## History

* @klebanov -- (`NEW_BUG`) _2013-12-20_

* @mulbrich -- (`BUGNOTE_ADDED`) _2013-12-20_

* @klebanov -- (`BUGNOTE_ADDED`) _2013-12-20_

* @rbubel -- (`BUGNOTE_ADDED`) _2013-12-20_

* @klebanov -- (`BUGNOTE_ADDED`) _2013-12-20_

* @klebanov -- (`BUGNOTE_ADDED`) _2013-12-20_

* @rbubel -- (`NORMAL_TYPE`) _2013-12-21_

* @rbubel -- (`NORMAL_TYPE`) _2013-12-21_

* @rbubel -- (`NORMAL_TYPE`) _2013-12-21_

* @klebanov -- (`NORMAL_TYPE`) _2013-12-21_

* @grahl -- (`NORMAL_TYPE`) _2015-09-28_

## Attributes

* Category: Deployment
* Status: CLOSED
* Severity: MAJOR
* OS: 
* Target Version: 
* Resolution: WONT_FIX
* Priority: NORMAL
* Reproducibility: ALWAYS
* Platform: 
* Commit: e24873c987061634157b3e8d2ec02b06374e8ea1
* Build: 
* Tags []
* Labels:  ~Deployment  ~Bug  ~NORMAL  

---

[View in Mantis](http://i12www.ira.uka.de/~klebanov/mantis/view.php?id=1391)

                ----

                Information: 
                    * created_at: 2017-05-29T02:55:21.743Z
                    * updated_at: 2017-05-29T02:55:22.491Z
                    * closed_at: 2017-05-29T02:55:22.446Z  (closed_by: )
                    * milestone: 
                    * user_notes_count: 0
